packages/contracts/contracts/DependencyRegistryV0.sol

```diff
@@ -53,617 +53,662 @@ contract DependencyRegistryV0 is
-        _onlyExistingLicenseType(_licenseType);
+        _onlyExistingLicenseType(licenseType);
         require(
-            _dependencyType.containsExactCharacterQty(
+            dependencyNameAndVersion.containsExactCharacterQty(
                 AT_CHARACTER_CODE,
                 uint8(1)
             ),
             "must contain exactly one @"
         );
         require(
             // @dev the add function returns false if set already contains value
-            _dependencyTypes.add(_dependencyType),
-            "Dependency type already exists"
+            _dependencyNameVersionIds.add(dependencyNameAndVersion),
+            "Dependency already exists"
         );
 
-        Dependency storage dependency = dependencyDetails[_dependencyType];
-        dependency.licenseType = _licenseType;
-        dependency.preferredCDN = _preferredCDN;
-        dependency.preferredRepository = _preferredRepository;
-        dependency.referenceWebsite = _referenceWebsite;
+        Dependency storage dependency = dependencyRecords[
+            dependencyNameAndVersion
+        ];
+        dependency.licenseType = licenseType;
+        dependency.preferredCDN = preferredCDN;
+        dependency.preferredRepository = preferredRepository;
+        dependency.website = dependencyWebsite;
 
         emit DependencyAdded(
-            _dependencyType,
-            _licenseType,
-            _preferredCDN,
-            _preferredRepository,
-            _referenceWebsite
+            dependencyNameAndVersion,
+            licenseType,
+            preferredCDN,
+            preferredRepository,
+            dependencyWebsite
         );
     }
 
     /**
      * @notice Removes a dependency.
-     * @param _dependencyType Name of dependency type (i.e. "type@version") used to identify dependency.
+     * @param dependencyNameAndVersion Name and version of dependency (i.e. "name@version") used to identify dependency.
      */
-    function removeDependency(bytes32 _dependencyType) external {
+    function removeDependency(bytes32 dependencyNameAndVersion) external {
         _onlyAdminACL(this.removeDependency.selector);
-        _onlyExistingDependencyType(_dependencyType);
-        Dependency storage dependency = dependencyDetails[_dependencyType];
+        _onlyExistingDependency(dependencyNameAndVersion);
+        Dependency storage dependency = dependencyRecords[
+            dependencyNameAndVersion
+        ];
         require(
             dependency.additionalCDNCount == 0 &&
                 dependency.additionalRepositoryCount == 0 &&
                 dependency.scriptCount == 0,
             "Cannot remove dependency with additional CDNs, repositories, or scripts"
         );
 
-        _dependencyTypes.remove(_dependencyType);
-        delete dependencyDetails[_dependencyType];
+        _dependencyNameVersionIds.remove(dependencyNameAndVersion);
+        delete dependencyRecords[dependencyNameAndVersion];
```

Comment:
Want to double-click that this won't delete the mapping items in a `Dependency` struct. I think this is fine, since they are limited to `additionalCDNs`, `additionalRepositories`, and `scriptBytecodeAddresses `.
I definitely worth a dev comment in the function and in-line here though!
Alternatively, we could loop through these items and delete them, as this is a managed contract, and they should never get near a block gas limit to perform the looping operation.

Reply:
@ryley-o we have the require above to check that those arrays are empty before this function can be called so I think it's okay as is. Let me know if you think I should remove the require and delete them in this function instead.

Reply:
ohhhh nice! maybe an in-line dev comment right above this would be helpful to remind anyone reading!